File: Code/Websites/DanpheEMR/wwwroot/DanpheApp/src/app/labs/lab-settings/shared/lab-test-component-map.model.ts
import * as moment from 'moment/moment';
import { ENUM_DateTimeFormat } from '../../../shared/shared-enums';
import { LabComponentModel } from '../../shared/lab-component-json.model';

export class LabTestComponentMap {

    public ComponentMapId: number = 0;
    public LabTestId: number = 0;
    public DisplaySequence: number = 100;
    public ComponentId: number = 0;
    public IndentationCount: number = 0;
    public ShowInSheet: boolean = true;
    public CreatedOn: string = moment().format(ENUM_DateTimeFormat.Year_Month_Day_Hour_Minute);
    public CreatedBy: number = 0;
    public ModifiedOn: string = null;
    public ModifiedBy: number = null;

    public IsActive: boolean = true;
    public IsAutoCalculate: boolean = false;

    public CalculationFormula: string = "";

    public FormulaDescription: string = "";

    public LabTestComponent: LabComponentModel = null;
}

File: Code/Components/DanpheEMR.ServerModel/LabModels/LabTestComponentMapModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DanpheEMR.ServerModel.LabModels
{
    public class LabTestComponentMapModel
    {
        [Key]
        public int ComponentMapId { get; set; }
        public Int64 LabTestId { get; set; }
        public int ComponentId { get; set; }
        public int DisplaySequence { get; set; }
        public int? CreatedBy { get; set; }
        public DateTime? CreatedOn { get; set; }
        public int? ModifiedBy { get; set; }
        public DateTime? ModifiedOn { get; set; }

        public bool IsActive { get; set; }

        public string GroupName { get; set; }

        public int IndentationCount { get; set; }
        public bool? ShowInSheet { get; set; }
    }
}
